Feature/3 add react features handling

leosheecool opened the pull request:
https://github.com/Ealab-collab/reactive-json/issues/3

leosheecool committed:
feat(3): add environment variables, url and storage management.
Update dependencies in package-lock.json and refactor template evaluation logic. Introduced utility functions for data location checks and reactive JSON features. Enhanced the Switch component to utilize new utility functions, improving data handling. Added new examples in demo.yaml for reactive JSON usage.

Dakwamine commented on lib/engine/TemplateSystem.jsx:
```diff
@@ -352,18 +375,18 @@ export const isTemplateValue = (valueToEvaluate) => {
  * @constructor
  */
-const TemplateValue = ({ valueToEvaluate }) => {
+const templateValue = ({ valueToEvaluate }) => {
```
This should be a react component that is used in View.jsx, not a simple function. The first letter should be upper cased.

leosheecool replied:
This function returns only values and never JSX. Therefore, it shouldn't be a React component but a custom hook. I will change the name to `useTemplateValue`.

Dakwamine commented:
Thanks! I still did not test the pull request thoroughly yet I see a lot of good ideas in it!

Here are my thoughts:

- better function naming for the pattern detection system:
  - utilsRegex.jsx => placeholderPattern.jsx or something like that (keep it singular)
  - isReactiveJsonReactFeature, isGetLocalStorageFeature, and others => I don't understand what they really do. Can you add comments?
- when it's done, we can break the utilsRegex.jsx into multiple files, at least the system core, and the implementations.
  - the main objective is to allow the user to plug new patterns, or even, remove unwanted ones. You should use the reactive-json plugin system to make things easy.

leosheecool committed:
refactor(3): change naming and folder architecture

leosheecool committed:
fix(3): rebase and fix rebase issues

leosheecool committed:
feat(3): implement cookie management in reactive JSON. Added support for setting, getting, and removing cookies through new placeholders.
Updated demo.yaml with examples for cookie usage. Enhanced utility functions for cookie parsing and management.
TODO: add dev mode management in the manageCookies() function to handle the fallback to Lax only in dev.

leosheecool committed:
refactor(3): rename templateValue to TemplateValue
